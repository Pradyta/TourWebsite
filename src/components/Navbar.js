import { Component } from "react";
import "./cssFiles/NavbarStyle.css";
import { Outlet, Link } from "react-router-dom";

class Navbar extends Component {
  render() {
    return (
      <>
        <nav>
          <h1 id="logo">Trip Advisor</h1>

          <div>
            <ul id="navbar">
              <li>
                <i className="fa fa-home" aria-hidden="true"></i>
                <Link to="/">Home</Link>
              </li>

              <li>
                <i className="fa-solid fa-info"></i>
                <Link to="/about">About</Link>
              </li>

              <li>
                {" "}
                <i className="fa-solid fa-address-book"></i>
                <Link to="/contact">Contact Us</Link>
              </li>

              <button>
                <Link to="/login">Login</Link>
              </button>

              <button>
                <Link to="/signUp">Sign Up</Link>
              </button>
              
            </ul>
          </div>
        </nav>

        <Outlet />
      </>
    );
  }
}

export default Navbar;
