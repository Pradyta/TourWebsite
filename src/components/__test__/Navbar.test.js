import { render, screen } from '@testing-library/react';
import Navbar from '../Navbar';
import {BrowserRouter as Router} from 'react-router-dom';

describe('Navbar Component', () => {
    test('should render Navbar compoennt', () => {
    const {asFragment} = render( 
        <Router>
        <Navbar />,
        </Router>
    );
    expect(asFragment(<Navbar/>)).toMatchSnapshot();
    });
    
});