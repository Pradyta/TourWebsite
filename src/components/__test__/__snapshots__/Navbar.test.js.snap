// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render Navbar compoennt 1`] = `
<DocumentFragment>
  <nav>
    <h1
      id="logo"
    >
      Trip Advisor
    </h1>
    <div>
      <ul
        id="navbar"
      >
        <li>
          <i
            aria-hidden="true"
            class="fa fa-home"
          />
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <i
            class="fa-solid fa-info"
          />
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
           
          <i
            class="fa-solid fa-address-book"
          />
          <a
            href="/contact"
          >
            Contact Us
          </a>
        </li>
        <button>
          <a
            href="/login"
          >
            Login
          </a>
        </button>
        <button>
          <a
            href="/signUp"
          >
            Sign Up
          </a>
        </button>
      </ul>
    </div>
  </nav>
  ,
</DocumentFragment>
`;
